### PR TITLE
Modern ObjC conversion and modernization

### PR DESCRIPTION
**Pull request by JanX2**

### CODE REVIEW

**Comment by JanX2** [comment]
This might require the travis changes by @josh64x2.

**Comment by josh64x2** [comment]
Thanks for this! Will review properly today but it does look good so far. Join `#ViennaRSS` on freenode if you can. :smile:

File: Vienna.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -7112,6 +7112,7 @@
 				CLANG_WARN_UNREACHABLE_CODE = YES;
 				CLANG_WARN__DUPLICATE_METHOD_MATCH = YES;
 				ENABLE_STRICT_OBJC_MSGSEND = YES;
+				GCC_NO_COMMON_BLOCKS = YES;
```
Comment (dak180) [inline Vienna.xcodeproj/project.pbxproj]: These changes should be in the `configs/Project-All.xcconfig` file rather than the project file itself.

**Comment by JanX2** [comment]
@josh64x2 @dak180 The additional commits should resolve the issues mentioned above. :)

**Comment by josh64x2** [comment]
@JanX2 Please check https://github.com/JanX2/vienna-rss-1/pull/1 to make sure I've fixed the merge conflicts successfully :smile:

**Comment by barijaona** [comment]
I am unsure if the "previously hidden bug" as described at commit de764874ec638071b20d4f4c95ddc067df63dc8c is really related to the new syntax…

**Comment by barijaona** [comment]
When I switch View -> Layout to Horizontal (the default value), I consistently get an exception : `NSConcreteMutableAttributedString initWithString:attributes:: nil value`.
This is triggered by something in the method `-[ArticleListView tableView:objectValueForTableColumn:row:]`

No problem with horizontal and vertical layouts.
Is one of you able to reproduce ?

**Comment by josh64x2** [comment]
@barijaona are you using this PR or https://github.com/JanX2/vienna-rss-1/pull/1 ?
I haven't been able to get your error but I have been getting these two when switching to or from the unified layout randomly:
`Row 0 out of row range [0--1] for rowViewAtRow:createIfNeeded:`
`Webview error associated to object of class (null)`

**Comment by barijaona** [comment]
The problem is with the original PR
Your merged branch seems better, but I have just begun my tests.

**Comment by josh64x2** [comment]
@barijaona I propose closing this and accepting #598 so we can move forward with other changes.

**Comment by barijaona** [comment]
Superseded by PR #598

**Pull request closed without merging**